### ISSUE TITLE
.go('back') not working the same after upgrade form 0.19 to 1.0.3

### ISSUE THREAD

**Issue by chriskilmer** [cypress-io/cypress#892]
- Operating System: Mac
- Cypress Version: 1.0.3
- Browser Version: Chrome  61.0.3163.100

### Is this a Feature or Bug?
Bug.

### Current behavior:
![image](https://user-images.githubusercontent.com/624/32578529-cf6bce30-c49a-11e7-8e86-654c349ef277.png)

If I look at `cy.hash()` after `cy.go('back')`, the hash is empty.

If I look at `cy.location()` after `cy.go('back')`,  I get `about:blank`.

### Desired behavior:
This was a passing spec before upgrading from 0.19 to 1.0.3

#### Test code:
This was a passing spec before upgrading from 0.19 to 1.0.3.

```js
cy
  .visit('/#/explore?zoom=4&lat=34.66936&lng=36.82617')
  .visit('/#/explore?zoom=9&lat=36.753537&lng=3.058833')
  .get('.leaflet-marker-icon')
  .should('have.length', 1)
  .go('back')
  .get('.leaflet-marker-icon')
  .should('have.length', 5)
  .go('forward')
  .get('.leaflet-marker-icon')
  .should('have.length', 1)
})
```

**Comment by jennifer-shehane** [comment]
Hi, I have been looking into this issue some. I did find an issue with `cy.go()`, but it does not seem to be the same issue you are having: #896

Have you verified that this behavior works in your application independently outside of Cypress? That is, after visiting `/#/explore?zoom=4&lat=34.66936&lng=36.82617` then visiting `/#/explore?zoom=9&lat=36.753537&lng=3.058833` and clicking 'back' in your browser that it indeed does go back to the correct previous hash?

I wrote a test similar to yours in my React application that does indeed go back correctly with the hash and also has the hash accessible from `cy.location()` and `cy.hash()`.

To investigate further, I'd need a more reproducible example (with an application I can visit) to recreate this exact bug.
